src/config_parser.c

```diff
@@ -37,49 +42,116 @@ static char *trim(char *s)
     return rtrim(ltrim(s));
 }
 
-void free_buffer(char **buffer)
+static bool is_key(const char *buffer, const char *key, int buffer_len)
+{
+  int ret = true;
+  char *raw_key, *parsed_key = NULL;
+  raw_key = strndup(buffer,buffer_len);
+  parsed_key = strdup(trim(raw_key)); /* strip whitespaces */
+  free(raw_key);
+  if (strcmp(parsed_key,key) != 0)
+    ret = false;
+  free(parsed_key);
+  return ret;
+}
+
+static void free_buffer(char **buffer)
 {
   free(*buffer);
 }
 
+/** @brief Set priority for a binary name in the given state struct.
+ *
+ * @param priority Priority which has to be set. The entry will be deleted
+                   if priority is <= 0.
+ * @param state Reference on an ConfigParserState object in which the
+ *              priority will be set.
+ * @return Complete string buffer (with other values)
+ */
+static const char *setBinaryPriority(int priority, struct ConfigParserState *state)
```

Comment:
this function very heavy, looks like O(n^2) for number of entries, but we don't really care.